kolibri/plugins/learn/assets/test/views/library-and-channel-browser-main-content-grid.spec.js

```diff
@@ -0,0 +1,68 @@
+import { shallowMount } from '@vue/test-utils';
+import LibraryAndChannelBrowserMainContent from '../../src/views/LibraryAndChannelBrowserMainContent';
+
+describe('Library and Channel Browser Main Content', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = shallowMount(LibraryAndChannelBrowserMainContent, {
+      computed: { windowIsSmall: () => false, backRoute: 'test' },
+      propsData: { contents: [{ node: 1 }], currentCardViewStyle: 'card' },
+    });
+  });
+  it('smoke test', () => {
+    expect(wrapper.exists()).toBe(true);
+  });
+  describe('When the user has a medium or large screen', () => {
+    describe('When `currentCardViewStyle` is a card', () => {
+      it('displays a `CardGrid`, and within the grid, a `HybridLearningContentCard` for each content node', () => {
+        expect(wrapper.find('[data-test="non-mobile-card-grid"]').element).toBeTruthy();
+        expect(wrapper.find('[data-test="content-card"]').element).toBeTruthy();
+      });
+      it('does not display a `CardGrid` with a `ResourceCard` for each content node in a single, full-width column', () => {
```

Comment:
This and some of the other tests like it doesn't test that "each content node" is rendered rather it only tests that the one in your propsData.contents does. Testing two would be relatively easy to do manually - you could try making each card's data-test dynamically generated `:data-test="'content-card-' + idx"` and then assert that they both show up. If you want to test more than two to be even safer, then `it.each` might help.

Reply:
Yup this is great! Thanks Marcella!